crates/astria-conductor/src/executor/client.rs

```diff
@@ -117,14 +117,14 @@ impl Client {
     pub(super) async fn execute_block_with_retry(
         &mut self,
         prev_block_hash: Bytes,
-        transactions: Vec<Vec<u8>>,
+        transactions: Vec<Bytes>,
         timestamp: Timestamp,
     ) -> eyre::Result<Block> {
         use prost::Message;
 
         let transactions = transactions
             .into_iter()
-            .map(|tx| RollupData::decode(tx.as_slice()))
+            .map(|tx| RollupData::decode(tx.as_ref()))
```

Comment:
I think you don't need to do that call here: just `map(RollupData::decode)` because `decode(impl Buf)`: https://docs.rs/prost/latest/prost/trait.Message.html#method.decode